scilpy/tracking/local_tracking.py

```diff
@@ -12,103 +12,97 @@
-                                              itertools.repeat(save_seeds))))
-                pool.close()
-                # Make sure all worker processes have exited before leaving
-                # context manager in order to prevent temporary file deletion
-                # errors in Windows
-                pool.join()
-                lines =\
-                    np.array([line for line in
+    # Verifying the number of processes
+    if nbr_processes <= 0:
+        try:
+            nbr_processes = multiprocessing.cpu_count()
+        except NotImplementedError:
+            warnings.warn("Cannot determine number of cpus. \
+                returns nbr_processes set to 1.")
+            nbr_processes = 1
+
+    if nbr_processes > params.nbr_seeds:
+        nbr_processes = params.nbr_seeds
+        logging.debug('Setting number of processes to ' +
+                      str(nbr_processes) +
+                      ' since there were less seeds than processes.')
+    if nbr_processes < 2:
+        chunk_id = 1
+        lines, seeds = get_streamlines(
+            tracker, mask, seed_generator, chunk_id, params, compression_th,
+            nbr_processes=1, save_seeds=save_seeds)
+    else:
+        # Each process will use get_streamlines_at_seeds
+        chunk_ids = np.arange(nbr_processes)
+        with nib.tmpdirs.InTemporaryDirectory() as tmpdir:
+            # toDo
+            # must be better designed for dipy
+            # the tracking should not know which data to deal with
+            data_file_name = os.path.join(tmpdir, 'data.npy')
+            np.save(data_file_name, tracker.tracking_field.dataset.data)
+            tracker.tracking_field.dataset.data = None
+
+            pool = multiprocessing.Pool(nbr_processes,
+                                        initializer=_init_sub_process,
+                                        initargs=(data_file_name,
+                                                  params.mmap_mode))
+
+            lines_per_process, seeds_per_process = zip(*pool.map(
+                _get_streamlines_sub,
+                zip(itertools.repeat(tracker),
+                    itertools.repeat(mask),
+                    itertools.repeat(seed_generator),
+                    chunk_ids,
+                    itertools.repeat(params),
+                    itertools.repeat(compression_th),
+                    itertools.repeat(nbr_processes),
+                    itertools.repeat(save_seeds))))
+            pool.close()
+            # Make sure all worker processes have exited before leaving
+            # context manager in order to prevent temporary file deletion
+            # errors in Windows
+            pool.join()
+            lines = np.array([line for line in
                               itertools.chain(*lines_per_process)])
-                seeds =\
-                    np.array([seed for seed in
+            seeds = np.array([seed for seed in
```

Comment:
I don't think `lines` and `seeds` need to be numpy arrays? It raises a `DeprecationWarning` because it is an array of lists. We could just remove the englobing `np.array`.